src/tribler-core/tribler_core/modules/popularity/test_popularity_community.py

```diff
@@ -75,6 +75,51 @@ async def test_torrents_health_gossip(self):
             assert torrent.leechers == checked_torrent_info[2]
             assert torrent.last_check == checked_torrent_info[3]
 
+    async def test_torrents_health_gossip_multiple(self):
+        """
+        Test whether torrent health information is correctly gossiped around
+        """
+        # torrent structure is (infohash, seeders, leechers, last_check)
+        dead_torrents = {(random_infohash(), 0, randint(1, 10), int(time.time()))
+                         for _ in range(10)}
+        alive_torrents = {(random_infohash(), randint(1, 10), randint(1, 10), int(time.time()))
+                          for _ in range(PopularityCommunity.GOSSIP_RANDOM_TORRENT_COUNT)}
+        top_popular_torrents = {(random_infohash(), randint(11, 100), randint(1, 10), int(time.time()))
+                                for _ in range(PopularityCommunity.GOSSIP_POPULAR_TORRENT_COUNT)}
+
+        all_checked_torrents = dead_torrents | alive_torrents | top_popular_torrents
+
+        db1 = self.nodes[0].overlay.mds.TorrentState
+        db2 = self.nodes[1].overlay.mds.TorrentState
+
+        with db_session:
+            assert db1.select().count() == 0
+            assert db2.select().count() == 0
+
+        for torrent_info in all_checked_torrents:
+            self.nodes[0].overlay.torrent_checker.torrents_checked.add(torrent_info)
+
+        await self.introduce_nodes()
+
+        # Gossip random torrents
+        self.nodes[0].overlay.gossip_random_torrents_health()
+        await self.deliver_messages(timeout=0.1)
+
+        # Check whether node 1 has received all random torrent health information
```

Comment:
db1 == node0 and db2 == node1 naming scheme is inconsistent and surprising. It is better to use the same index everywhere, eg db0, db1, or even node_sender/db_sender, node_receiver/db_receiver.

Reply:
this test and the naming scheme is an extension to the existing test `test_torrents_health_gossip` (line 56). I can change the naming scheme in both of them if you prefer.

Reply:
Yep, that would be superb! :+1: